ckanext/recombinant/logic.py

```diff
@@ -263,14 +264,19 @@ def _update_datastore(lc, geno, dataset, force_update=False):
 
 
 def _update_triggers(lc, chromo):
-    field_choices = {}
+    definitions = {}
     trigger_names = []
 
+    if u'trigger_strings' in chromo:
+        definitions = chromo['trigger_strings']
+
     for f in chromo['fields']:
         if 'choices' in f:
-            field_choices[f['datastore_id']] = sorted(f['choices'])
+            assert f['datastore_id'] not in definitions, '{} already defined in trigger_strings'.format(f['datastore_id'])
```

Comment:
we could add a prefix/suffix to choices and definitions so that they never conflict (and cause this error that might be confusing to users)

Reply:
@wardi as they are all user defined stuff in the yaml files, how would we make sure that a user uses the prefixes?

Reply:
good point. How about preventing use of field names as trigger_strings keys and having the error there instead?

Reply:
or update these errors to say "trigger_string {name} can't be used because that name is required for the {name} field choices"

Reply:
@wardi errors have been updated with more useful copy. And we are now raising exceptions here too.